Feature/cypress mongo

dacort opened the pull request:
Implements #12286
Reproduces #12265

A few comments:
- This is the first step in allowing us to have database-specific Cypress tests
- I added a new executor `fe-mongo` that combines `clojure-and-node` and `mongo`
- I updated the `fe-tests-cypress` job to take a set of parameters that allow it to tweak the Cypress command
- I updated the `build-uberjar` job to also build all the drivers
- I updated `run_cypress_tests.js` to take a new argument that only runs a specific set of tests
- I updated our `cypress.json` to point one path deeper in order to accommodate the above
- Added two new MongoDB-specific tests, one of which will fail until #12265 is fixed

dacort committed:
Initial changes to Cypress to allow a limited set of tests to run

dacort committed:
Move Cypress DB tests around and update circle executor

dacort committed:
Include drivers in frontend uberjar

paulrosenzweig commented on .circleci/config.yml:
```diff
@@ -560,11 +572,24 @@ jobs:
+      test-files-location:
+        type: string
+        default: ""
+      driver:
```
Where is this used?

dacort replied:
It’s used as part of the `run_cypress_tests` below. So there’s a Boolean flag to enable the database specific tests and then this passes in a path that those tests live in. Currently this is only `frontend/test/metabase-db/mongo` but will eventually be other DB types.

paulrosenzweig reviewed (approved):
🎉 This is awesome!

- It might make sense to have separate snapshots for different drivers' tests. That doesn't really matter until we have a lot more tests and will be a lot easier with #12242 done.
- What's the best way to develop these tests locally? I imagine the mongo image has some pre-seeded db? Have you been running that image locally? Or, is there some script that is used to seed that db?

dacort commented on frontend/test/metabase-db/mongo/add.cy.spec.js:
```diff
@@ -0,0 +1,60 @@
+  cy.findByText("Save")
+    .should("not.be.disabled")
+    .click();
+}
```
Yea I think that makes sense. We should have one canonical test for the UI, but then other setup for tests should probably happen via the API.

dacort replied:
Updated the PR to utilize this approach.

dacort commented:
> - What's the best way to develop these tests locally? I imagine the mongo image has some pre-seeded db? Have you been running that image locally? Or, is there some script that is used to seed that db?

That’s something we still need to figure out. :/ Right now this is just an empty DB. We could potentially have our own images with seed data and I think this would be the way to go.

dacort committed:
Addressing a few PR review comments:
- Move type field function into Cypress support file
- Create database using API call rather than UI in repetitive tests
- Remove it.only statement 😬

dacort commented:
May make one more change to this PR to separate out `build-uberjar` into one that has drivers and one that does not. The driver build takes significantly longer (doubles the build time from ~7m to ~15?).

dacort commented:
> May make one more change to this PR to separate out `build-uberjar` into one that has drivers and one that does not.

Leaving as is for now - whether we have one or two, the longer one will have to build in any case the shorter one does. If this becomes an issue, we can address at that point.
